BraveWallet/Crypto/Scanner/AddressQRCodeScannerView.swift

```diff
@@ -0,0 +1,234 @@
+      .navigationBarTitleDisplayMode(.inline)
+      .toolbar {
+        ToolbarItemGroup(placement: .cancellationAction) {
+          Button(action: {
+            presentationMode.dismiss()
+          }) {
+            Text(Strings.CancelString)
+              .foregroundColor(Color(.braveOrange))
+          }
+        }
+      }
+      .ignoresSafeArea()
+      #else
+      _AddressQRCodeScannerView(address: $address,
+                                isErrorPresented: $isErrorPresented,
+                                isPermissionDenied: $permissionDenied,
+                                dismiss: { presentationMode.dismiss() }
+      )
+        .navigationBarTitleDisplayMode(.inline)
+        .toolbar {
+          ToolbarItemGroup(placement: .cancellationAction) {
+            Button(action: {
+              presentationMode.dismiss()
+            }) {
+              Text(Strings.CancelString)
+                .foregroundColor(Color(.braveOrange))
+            }
+          }
+        }
+        .ignoresSafeArea()
+        .background(
+          Color.clear
+            .alert(isPresented: $isErrorPresented) {
+              Alert(
+                title: Text(""),
+                message: Text(Strings.scanQRCodeInvalidDataErrorMessage),
+                dismissButton: .default(Text(Strings.scanQRCodeErrorOKButton), action: {
+                  presentationMode.dismiss()
+                })
+              )
+            }
+        )
+        .background(
+          Color.clear
+            .alert(isPresented: $permissionDenied) {
+              Alert(
+                title: Text(""),
+                message: Text(Strings.scanQRCodePermissionErrorMessage),
+                dismissButton: .default(Text(Strings.openPhoneSettingsActionTitle), action: {
+                  UIApplication.shared.open(URL(string: UIApplication.openSettingsURLString)!)
+                })
+              )
+            }
+        )
+      #endif
+    }
+  }
+}
+
+private struct _AddressQRCodeScannerView: UIViewControllerRepresentable {
+  typealias UIViewControllerType = AddressQRCodeScannerViewController
+  @Binding var address: String
+  @Binding var isErrorPresented: Bool
+  @Binding var isPermissionDenied: Bool
+  var dismiss: () -> Void
+  
+  func makeUIViewController(context: Context) -> UIViewControllerType {
+    AddressQRCodeScannerViewController(address: _address,
+                                isErrorPresented: _isErrorPresented,
+                                isPermissionDenied: _isPermissionDenied,
+                                dismiss: self.dismiss
+    )
+  }
+  
+  func updateUIViewController(_ uiViewController: UIViewControllerType, context: Context) {
+  }
+}
+
+private class AddressQRCodeScannerViewController: UIViewController {
+  @Binding private var address: String
+  @Binding private var isErrorPresented: Bool
+  @Binding private var isPermissionDenied: Bool
+  
+  private let captureDevice = AVCaptureDevice.default(for: .video)
```

Comment:
Any reason we didn't use: `SyncCameraView`?

Reply:
`SyncCameraView` is in `Client`, no way to import it, I guess didn't want to put it in BraveShared?